processing/src/main/java/io/druid/segment/MetricHolder.java

```diff
@@ -109,6 +106,9 @@ public static void writeToChannel(MetricHolder holder, WritableByteChannel out)
     serializerUtils.writeString(out, holder.typeName);
 
     switch (holder.type) {
+      case LONG:
```

Comment:
This would still be worth bringing up in a separate PR, and bonus points if you can spot the unit test which missed it.